### ISSUE TITLE
validateSocksClientOptions unhandled exception

### ISSUE THREAD

**Issue by liteCarma** [JoshGlazebrook/socks#61]
If the callback function is used, this causes an unhandled exception.
```javascript
     SocksClient.createConnection(options, (err, info) => {


      })
```
I had to do this
```javascript
    try {
      SocksClient.createConnection(options, (err, info) => {

      })
    } catch (err) {
     
    }
```

**Comment by JoshGlazebrook** [comment]
This is kind of as-expected. The error returned via the callback is more for errors that occur during the proxy connection setup. An error thrown from validateSocksClientOptions means something is wrong with your connection options.

**Comment by liteCarma** [comment]
> 
> 
> This is kind of as-expected. The error returned via the callback is more for errors that occur during the proxy connection setup. An error thrown from validateSocksClientOptions means something is wrong with your connection options.

I don't think this is how it should work. You have an example of this behavior of the built-in node.js  functionswith similar behavior? In the end, it looks bad.

**Comment by JoshGlazebrook** [comment]
This is fixed in 2.5.0 - https://github.com/JoshGlazebrook/socks/commit/40890dc06800994af2513ce44fd21bc25fe2362e
